core/load_copyright.py

```diff
@@ -1,23 +1,43 @@
 import logging
 import sys
 from core.models import Copyright
+from django.db import transaction
+from django.core.validators import URLValidator
+from django.core.exceptions import ValidationError
 
 def loadCopyright(inputfilename):
 
     _logger = logging.getLogger(__name__)
+    val = URLValidator()
 
     try:
         f = open(inputfilename, 'r')
-
-        for line in f:
-            arr = line.split("\t")
-            if len(arr) == 2:
-              record = Copyright()
-              record.uri = arr[0]
-              record.label =arr[1].strip()
-              record.save()
-    except Exception as e:
+    except IOError as e:
         _logger.exception(e)
+        sys.exit(1)
+
+    for line in f:
+        arr = line.split("\t")
+        if len(arr) == 2:
+            try:
+                val(arr[0])
+            except ValidationError as e:
+                _logger.exception(e)
+                sys.exit(1)
+            record = Copyright()
+            record.uri = arr[0]
+            record.label =arr[1].strip()
+            try:
+                with transaction.atomic():
```

Comment:
Could you move this `with` statement to wrap the entire `for` block?  We want to make sure all the entries are valid or else nothing gets in the database.